addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -736,11 +738,20 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
     willStart: async function () {
         const _super = this._super.bind(this);
         // Get the authorized existing fields for the form model
-        this.existingFields = await this._rpc({
-            model: "ir.model",
-            method: "get_authorized_fields",
-            args: [this.formEl.dataset.model_name],
-        }).then(fields => {
+        const model = this.formEl.dataset.model_name;
+        let fields;
+        if (model in authorized_fields_cache){
+            fields = authorized_fields_cache[model];
+        } else {
+            fields = await this._rpc({
+                model: "ir.model",
+                method: "get_authorized_fields",
+                args: [this.formEl.dataset.model_name],
+            });
+        }
+
+        this.existingFields = (fields => {
+            authorized_fields_cache[model] = fields;
```

Comment:
It's really nice to try and preserve the git history but here I would reindent properly instead of having a strange anonymous function for no apparent reason :wink: Code quality is more important than easier git blame calls especially for 10 lines :wink: 

Here you could also "cache" the `this.fields` part: your cache could be a map <model name> -> Promise resolving to an array/object containing `existingFields` and `fields`. That way everything that can be precomputed actually is. But that's not really important, just caching the RPC result is fine too (as you wish).